File: 06/assembler/label_parser.rb
require_relative 'pattern.rb'
require_relative 'parser.rb'

class LabelParser < Parser
  def initialize(filename, symbol_map)
    super
    @next_addr = 0 # ROM address
  end

  private

  def parse_line(line)
    label_match = line.match(Pattern.label)
    if label_match
      @symbol_map[label_match[1]] = @next_addr
    else
      @next_addr += 1
    end
  end
end

File: 06/assembler/src/tecs/assembler/LabelParser.java
//package tecs.assembler;
//
//import java.nio.file.Path;
//import java.util.regex.Matcher;
//import java.util.regex.Pattern;
//
//class LabelParser extends Parser {
//    private int nextRomAddress;
//
//    LabelParser(Path filePath, SymbolTable table) {
//        super(filePath, table);
//    }
//
//    protected void parseLine(String line) {
//        Matcher m = Pattern.compile(AsmPattern.label).matcher(line);
//        if (m.matches()) {
//            table.addEntry(m.group(1), nextRomAddress);
//        } else {
//            nextRomAddress += 1;
//        }
//    }
//}
